core/src/main/java/org/opensearch/sql/ast/tree/CloseCursor.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.sql.ast.tree;
+
+import lombok.Getter;
+import lombok.RequiredArgsConstructor;
+import org.opensearch.sql.ast.AbstractNodeVisitor;
+import org.opensearch.sql.ast.Node;
+import org.opensearch.sql.executor.pagination.PlanSerializer;
+
+import java.util.List;
+
+/**
+ * TODO
+ */
+public class CloseCursor extends UnresolvedPlan {
+
+  /** An instance of {@link Cursor} */
+  private UnresolvedPlan child;
```

Comment:
What is this child used for? 

As far as I can see, this can be cursor id without loss of generality and would make the code more straight-forward.

Reply:
True.
I tried to follow this graph:
![image](https://github.com/opensearch-project/sql/assets/88679692/ac90e7c0-4fd3-4a5c-98a0-26e6698e1d86)

If I store here the cursor itself instead of a child plan, the planner logic should be a bit more complicated to create 2 nodes.

Reply:
I see how it works but it's hard to follow. I hope we can make it more explicit at some point later.